README.md

```diff
@@ -1,62 +1,181 @@
-./bin/build_stack
 
-# Build everything in parallel
-docker-compose build --parallel
+# Start everything
+./bin/start
+```
+
+### Updating everything
+
+The first time you use `./bin/start` we will download the latest versions of all the Exercism components for you to use.
+When you want to upgrade those components, run:
+
+```bash
+docker-compose pull
+```
+
+If you face any issues getting started, we recommend running this step in case anything is cached locally from some previous installation.
+
+### Logs
+
+Once the docker stack is running you can tail the logs using:
 
-# This is your normal starting command
-docker-compose up --build
+```bash
+docker-compose logs -f
+```
+
+You can also tail just one component by specifying its name.
+For example, to tail the website logs:
+
+```bash
+docker-compose logs -f website
 ```
 
-## Hacking on certain components of the architecture
+### Shelling into a container
+
+From time to time you may want to shell into a running container to run commands or check what is going on.
+To shell into a running container you can use the provided `bin/shell` command.
+For example, to shell into the running `website` container, you would run:
+
+```bash
+bin/shell website
+```
+
+### Stopping everything
+
+To stop everything, run:
+```
+docker-compose down
+```
+
+## What is actually going on?
+
+### Docker Compose
 
-In order to hack on specific pieces of the architecture you'll need to fetch those from Git.  Say to work on tooling invoker.
+We are using [Docker Compose](https://docs.docker.com/compose/).
+Compose is a tool for defining and running multi-container Docker applications.
+With Compose, you use a YAML file to configure your application’s services.
+Then, with a single command, you create and start all the services from your configuration.
+
+In the context of Exercism, each component (e.g. the website UI, an analyzer, the code that manages test runners) has it's own Dockerfile.
+These Dockerfiles are stored within the repositories (often named `dev.Dockerfile`) and are built and pushed to DockerHub via GitHub Actions.
+
+The code in this repository handles the creation of a `docker-compose.yml` and provides you with some simple wrapper scripts to run things.
+The `bin/start` script generates the `docker-compose.yml` from your `stack.yml` file, downloads the images from DockerHub and then starts them (via `docker-compose up`).
+
+### stack.yml
+
+The `stack.yml` file is a Exercism-specific configuration file that allows you to select which components you want to run locally, and any configuration you want to do.
+We then take that configuration and do the hard-work of turning that into a `docker-compose.yml`.
+
+Every docker image you download takes up HDD space, and every docker container you run takes up memory, so the `stack.yml` file allows to you ensure that you only run the things that are necessary for whichever part of Exercism you are working on.
```

Comment:
> On that note, it might be a good idea to add a reminder to change the location of Docker's virtual disk so it doesn't fill up the main drive.

@SaschaMann Can you expand pls?

Reply:
Docker for Windows (not sure about other OS) has a setting for where it should store the virtual harddrives (? not sure about the details tbh). It defaults to C: which may not be what you want.

Reply:
I've added https://github.com/exercism/v3-docker-compose/issues/14